Add first draft of Shimmer introduction (#392)
iota-community/iota-wiki: internal/learn/future/shimmer.md

It can be used to work with upcoming IOTA updates before they arrive on the mainnet. It will also have its own token called [SMR](https://shimmer.network/token).
To fairly distribute tokens on the new network, you can stake your IOTA tokens before launch.

## Pre-launch staking

Before the Shimmer network goes online, the IOTA Foundation will announce a start date for "pre-launch" staking.
You will be able to make a transaction to yourself with a certain payload to declare that you want to stake that amount of IOTA tokens to receive SMR. Staking will now be possible for 90 days. You will earn 1 SMR for every 1Mi every 10 seconds.
Let's take a closer look at how this works.

### Technical overview

Those who are already familiar with the Build/Burn Vote process will recognize the similarities between the Vote and Staking, as it is performed by the same Hornet plugin.  
Hornet operators can enable this plugin. Hornet will then (after a certain milestone) start accumulating all outputs that have the specific Staking payload every 10 seconds. Until the 90 days are up.  

> Not clear enough? Let's use an example:  

Reviewer: ```suggestion
Let's see an example of how this would work:  
```

Could be perceived a bit passive aggressive

Author: Sounds good

---
id: shimmer
title: Shimmer
description: Introduction to the emerging incentivized staging network
keywords:
  - Incentivized
  - staking
  - SMR
---

IOTA Foundation [introduced](https://blog.iota.org/introducing-iota-staking/) Shimmer. An incentivized staging network.
It can be used to work with upcoming IOTA updates before they arrive on the mainnet. It will also have its own token called [SMR](https://shimmer.network/token).
To fairly distribute tokens on the new network, you can stake your IOTA tokens before launch.

## Pre-launch staking

> Before the Shimmer network goes online, the IOTA Foundation will announce a start date for "pre-launch" staking.
> You will be able to make a transaction to yourself with a certain payload to declare that you want to stake that amount of IOTA tokens to receive SMR. Staking will now be possible for 90 days. You will earn 1 SMR for every 1Mi every 10 seconds.
> There is no risk in staking your IOTA Tokens.

Reviewer: ```suggestion
During a 90 day period before the launch of the Shimmer network, you will be able to stake your IOTA tokens to receive SMR tokens on the Shimmer network upon launch. The IOTA Foundation will announce the start date 24 hours in advance.
During this 90 day period, you will earn 1 SMR for every 1Mi you staked every 10 seconds.
There is no risk in staking your IOTA tokens, you will remain in control of your tokens and won't have to transfer them to a third party.
```

What I tried to do here is describe the Pre-launch staking from a bit more abstract point of view and move the specifics down into the Technical overview.

Let's take a closer (simplified) look at the technology behind the process.

### Technical overview

> 

Reviewer: Most newcomers won't know what Hornet is, so maybe we need to explain this at a more abstract level first and move the specifics more down towards the end?

Author: Hmm towards the end of what? The paragraph or the article?

Reviewer: As in have the first part with the introduction to Shimmer and the example of Bob using his wallet to stake and after that go into specifics like the Hornet plugin, does that make sense? I think we could make the distinction between what a user needs to do and what the nodes will do for them a bit more clear.

Author: But I use the example to better explain the plugin too. So if I move the example up no one will get what the plugin is 😅

Reviewer: I see what you mean, one sec.

> Those who are already familiar with the Build/Burn Vote process will recognize the similarities between the Vote and Staking, as it is performed by the same Hornet plugin.  
> Hornet operators can enable this plugin. Hornet will then (after a certain milestone) start accumulating all outputs that have the specific Staking payload every 10 seconds. Until the 90 days are up.  

Reviewer: ```suggestion
Staking will be done through a plugin for the Hornet node, one of the node implementations of IOTA, which node operators can enable. Those who are already familiar with the upcoming Build/Burn Vote process will recognize the similarities between that voting process and the Shimmer staking process, as it is performed by the same Hornet plugin. To stake your IOTA tokens for SMR tokens, you will make a transaction to yourself with a certain payload to declare that you want to stake that amount of IOTA tokens to receive SMR tokens. The Hornet plugin will pick up these transactions and will start accumulating SMR tokens on the corresponding outputs as long as the output is not spent.
```

Is this somewhat correct?

Not clear enough? Let's use an example:  

> Bob wants to use the 1Mi in his wallet. He clicks the stake button in Firefly, which generates the transaction to himself with the specific payload. Hornet sees this message and starts accumulating the credit. So after a milestone, the value will be 1 SMR (remember that this token doesn't exist yet. It is just a number in a DB), after the second milestone in 10s you will have 2 SMR and so on. As long as your tokens stay where they are, you will accumulate "tokens".

Reviewer: ```suggestion
Bob has 1Mi on an address that he manages using Firefly, the official IOTA wallet. He wants to use his IOTA tokens to stake for some Shimmer tokens (SMR). He clicks the stake button in Firefly, which generates the transaction to himself with the specific payload. Hornet sees this message and starts accumulating the credit. So after a milestone, the value will be 1 SMR (remember that this token doesn't exist yet. It is just a number in a database). 10 seconds later, after the second milestone, he will have 2 SMR and so on. As long as his IOTA tokens stay on the address he used to stake, he will continue to accumulate SMR tokens.
```

> At the end of the 90 days, Hornet will create a file with all the "balances", which will be used to bootstrap the Shimmer network.  

Author: I need another word for file her 😅 I don't like that. I used output first, but then was afraid that it could get confused with output from UTXO. Maybe I should even use UTXO instead of output everywhere. What do you think?

Reviewer: Or just leave this technicality out and just say something like "at the end of the 90 days, Hornet will know how much SMR Bob is allowed to receive on the Shimmer network"?

Author: Yeah. Will do that

> At the end of the 90 days, Hornet will create a file with all the "balances", which will be used to bootstrap the Shimmer network.  

Reviewer: ```suggestion
At the end of the 90 days, Hornet will create a file with all the SMR balances, which will be used to bootstrap the Shimmer network.  
```

> On this network, you can now claim your real SMR tokens, which are stored at the same address as your IOTA tokens on the mainnet.  

Reviewer: ```suggestion
On the Shimmer network, Bob can now claim his real SMR tokens, which are stored at the same address as where his IOTA tokens are stores on the mainnet.
```

Author: stores? :trollface: 
And why did you remove the whitespaces at the end? I need them for a newline or did you want to remove that?

Reviewer: Fixed!

> At the end of the 90 days, Hornet will create a file with all the SMR balances, which will be used to bootstrap the Shimmer network.  

Author: ```suggestion
At the end of the 90 days, Hornet will know how much SMR tokens Bob is allowed to receive on the Shimmer network. This information will then be used to bootstrap the Shimmer network.  
```

Author: What do you think @jlvandenhout ? Still too technical?
I would like to keep it similar to this as I want to make it clear that Mainnet and Shimmer are not connected.

Reviewer: I think this is good.

> So the initial amount of SMR is defined by the amount of IOTA tokens staked.

Reviewer: ```suggestion
So the initial amount of SMR is defined by the amount of IOTA tokens staked and for how long they were staked during the 90 days before launch.
```
